### PR TITLE
Core `_io.concatenate()` may fail due to case when one of the cubes is scalar - this fixes that

### LINKED ISSUE #1716: _io.concatenate()` may fail if extracted laison cube is scalar
Raised (inadvertently) by @bouweandela in this [comment](https://github.com/ESMValGroup/ESMValCore/pull/1713#issuecomment-1236850683) - it all happens in `_io._concatenate_overlapping_cubes(cubes)` - after `extract_time()` the resulting cube may be scalar - so iris concatenation fails due to scalar cubes not having `dim_metadata` attributes in `iris._concatenate_CubeSignature` class (they do, but they are empty lists).

Now. my idea would be this to be fixed in `extract_time()` directly, but for now I'll fix it in our concatenation, not to belly up other preprocessors that use `extract_time()`.

Scalar cubes are PITAs!

Fixed in https://github.com/ESMValGroup/ESMValCore/pull/1715

### PR DESCRIPTION
**Pull request by valeriupredoi**
## Description

Closes #1716

The introduction of scalar cubes in iris complicates a number of things, including our advanced concatenation mechanism - it is possible, as seen from the test case changed in this PR, that along the path of concatenation as we do it, one of the processed cubes that we use as intermediary laison to be *scalar* - then the whole show crumbles. I am not a fan of scalar cubes as they are defined in iris, but that's for another day. Anyway, this PR fixes that particular corner case, and adds a test case (converts the previous test case to a more meaningful one, that is). 

All started from @bouweandela 's [comment](https://github.com/ESMValGroup/ESMValCore/pull/1713#issuecomment-1236850683) - cheers, Bouwe! :beer: 
***

## [Before you get started](https://docs.esmvaltool.org/projects/ESMValCore/en/latest/contributing.html#getting-started)

- [x] [☝ Create an issue](https://github.com/ESMValGroup/ESMValCore/issues) to discuss what you are going to do

## [Checklist](https://docs.esmvaltool.org/projects/ESMValCore/en/latest/contributing.html#checklist-for-pull-requests)

It is the responsibility of the author to make sure the pull request is ready to review. The icons indicate whether the item will be subject to the [🛠 Technical][1] or [🧪 Scientific][2] review.

[1]: https://docs.esmvaltool.org/en/latest/community/review.html#technical-review
[2]: https://docs.esmvaltool.org/en/latest/community/review.html#scientific-review

- [x] [🛠][1] This pull request has a [descriptive title and labels](https://docs.esmvaltool.org/projects/ESMValCore/en/latest/contributing.html#pull-request-title-and-label)
- [x] [🛠][1] [Unit tests](https://docs.esmvaltool.org/projects/ESMValCore/en/latest/contributing.html#tests) have been added
- [x] [🛠][1] Changes are [backward compatible](https://docs.esmvaltool.org/projects/ESMValCore/en/latest/contributing.html#backward-compatibility)
- [x] [🛠][1] All [checks below this pull request](https://docs.esmvaltool.org/projects/ESMValCore/en/latest/contributing.html#pull-request-checks) were successful

***

To help with the number pull requests:

- 🙏 We kindly ask you to [review](https://docs.esmvaltool.org/en/latest/community/review.html#review-of-pull-requests) two other [open pull requests](https://github.com/ESMValGroup/ESMValCore/pulls) in this repository

### CODE REVIEW

**Comment by valeriupredoi** [comment]
hold yer horses, I'll improve the test coverage - @bouweandela why is our test coverage test passing even if the coverage is shrinking? Go home test, you drunk :beer:

**Comment by bouweandela** [comment]
The test coverage test tests that all the changed code is covered by tests. So if you delete some code that is covered by tests from a file with less than 100% coverage (such as the debug log call you removed), the relative amount of covered code decreases, while the changed code is fully covered because there isn't any.

File: esmvalcore/preprocessor/_io.py [hunk 1]
```diff
@@ -501,9 +500,10 @@ def _concatenate_overlapping_cubes(cubes):
                                     data_start_1.month, data_start_1.day,
                                     start_overlap.year, start_overlap.month,
                                     start_overlap.day)
+            # convert c1_delta scalar cube to vector cube, if needed
+            if c1_delta.data.shape == ():
+                c1_delta = iris.util.new_axis(c1_delta, scalar_coord="time")
             cubes = iris.cube.CubeList([c1_delta, cubes[1]])
-            logger.debug("Attempting concatenatenation of %s with %s",
```
Comment (bouweandela) [inline esmvalcore/preprocessor/_io.py:505]: Did you want to remove this?

**Review by bouweandela — APPROVED** [review]

File: esmvalcore/preprocessor/_io.py (hunk 1 above)
Comment (valeriupredoi) [reply esmvalcore/preprocessor/_io.py:505]: nit at all, I was being silly - cheers for pointing it out, bud, will reinstate :+1:

**Comment by valeriupredoi** [comment]
> The test coverage test tests that all the changed code is covered by tests. So if you delete some code that is covered by tests from a file with less than 100% coverage (such as the debug log call you removed), the relative amount of covered code decreases, while the changed code is fully covered because there isn't any.

makes sense, cheers, Bouwe! V, go home, you're drunk :grin:

**Comment by valeriupredoi** [comment]
cheers for the review @bouweandela :beer: Your tests should all go fine now in your Dasky PR
